internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1554,6 +1554,12 @@ private CompletableFuture<PubSubProduceResult> sendMessage(
       PubSubProducerCallback callback,
       boolean updateDIV,
       PubSubMessageHeaders pubSubMessageHeaders) {
+    if (isClosed) {
+      CompletableFuture<PubSubProduceResult> future = new CompletableFuture<>();
```

Comment:
nit: You can just create a completed future instead. CompletableFuture.completedFuture(null);